### PR TITLE
O3-2820 Refactor the billing extension on start visit form

### PR DESCRIPTION
**Pull request by ODORA0**
## Requirements
- [ ] This PR has a title that briefly describes the work done including a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) type prefix and a Jira ticket number if applicable. See existing PR titles for inspiration.

#### For changes to apps
- [ ]  My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).

#### If applicable
At the moment all the attributes of this extension are specific for the KenyaEMR use case, this PR refactors the start visit form and adds configurations for hardcoded values
## Screenshots

https://github.com/openmrs/openmrs-esm-billing-app/assets/43242517/0a23a8b4-e97f-4741-8680-ed471430a3ed

### CODE REVIEW

File: src/billing-form/billing-checkin-form.component.tsx [hunk 1]
```diff
@@ -22,28 +22,20 @@ const BillingCheckInForm: React.FC<BillingCheckInFormProps> = ({ patientUuid, se
   const [paymentMethod, setPaymentMethod] = useState<any>();
   let lineList = [];
 
-  const shouldBillPatient =
```
Comment (pirupius) [inline src/billing-form/billing-checkin-form.component.tsx:25]: Please update the PR description and add any relevant screenshots or recording. I see you deleted `shouldBillPatient`

**Review by pirupius — CHANGES_REQUESTED** [review]

File: src/billing-form/billing-checkin-form.component.tsx (hunk 1 above)
Comment (ODORA0) [reply src/billing-form/billing-checkin-form.component.tsx:25]: I had to remove the hardcoded shouldBillPatient logic in billing checkin form based on the visit attributes component

File: src/config-schema.ts [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
 
+export interface BillingConfig {}
+
 export const configSchema = {
-  defaultCurrency: {
+  patientCatergory: {
+    _type: Type.Object,
+    _description: 'Encounter type UUIDs for TB.',
+    _default: {
+      paymentDetails: 'caf2124f-00a9-4620-a250-efd8535afd6d',
```
Comment (CynthiaKamau) [inline src/config-schema.ts:10]: perhaps we could separate the concepts from custom uuids from the visit attributes to make it easier to do the configuration

Comment (ODORA0) [reply src/config-schema.ts:10]: @CynthiaKamau Done

**Review by CynthiaKamau — APPROVED** [review]
LGTM

File: src/billing-form/billing-checkin-form.component.tsx [hunk 3]
```diff
@@ -20,11 +20,12 @@ const BillingCheckInForm: React.FC<BillingCheckInFormProps> = ({ patientUuid, se
   const { lineItems, isLoading: isLoadingLineItems, error: lineError } = useBillableItems();
   const [attributes, setAttributes] = useState([]);
   const [paymentMethod, setPaymentMethod] = useState<any>();
+  const { patientCatergory, catergoryConcepts } = useConfig();
```
Comment (pirupius) [inline src/billing-form/billing-checkin-form.component.tsx:23]: It looks like you have a typo on the word `category` and it affects other files as well.

**Review by pirupius — CHANGES_REQUESTED** [review]

**Review by pirupius — APPROVED** [review]
Will clean this up in a follow PR

**Pull request merged**